File: Códigos/Aula04.4.java
public class Clima{
    public String ano;
    public String mes;
    public String temperatura;
    public String precipitacao;

    public Clima(String ano, String mes, String temperatura, String precipitacao) {
        this.ano = ano;
        this.mes = mes;
        this.temperatura = temperatura;
        this.precipitacao = precipitacao;
    }

    @Override
    public String toString() {
        return "Dados climaticos(Ano: " + this.ano + ", Mês: " + this.mes + ", Temperatura: " + this.temperatura + ", Precipitação: " + this.precipitacao + ")";
    }
    //reescrevemos toString para exibir um objeeto completo da classe 
    @Override
    public boolean equals(Object obj) {
        Clima c = (Clima) obj;
        return (this.ano.equals(c.ano) && this.mes.equals(c.mes));
        //estamos dizendo quem são as chaves primárias (aquelas que não podem se repetir)
        //reescrevemos o metodo equals para definir os atributos identificadores
    }

    
    
}

File: Códigos/Aula04.4.py
class Clima:
    def __init__(self, ano, mes, temperatura, precipitacao): # type: ignore
        self.ano = ano
        self.mes = mes
        self.temperatura = temperatura
        self.precipitacao = precipitacao

    def __str__(self):
        return f"Dados climaticos - Ano: {self.ano}, Mês: {self.mes}, Temperatura: {self.temperatura}, Precipitação: {self.precipitacao}"

    def __eq__(self, object):

        if not isinstance(object, self.__class__):
            return NotImplemented
        return self.ano == object.ano and self.mes == object.mes
